staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -165,16 +184,23 @@ type RequestDigest struct {
 
 func (reqMgmt *requestManagementSystem) Wait(requestDigest RequestDigest) (execute bool, afterExecute func()) {
 	for {
-		rmState := reqMgmt.curState.Load().(*RequestManagementState)
+		rmState := reqMgmt.curState.Load().(*requestManagementState)
+
+		var hashValue uint64
+		var priorityLevelName = reqMgmt.globalDefaultPriorityLevelConfigurationName
 		fs := rmState.pickFlowSchema(requestDigest)
-		// RequestManagementState is constructed to guarantee that the following succeeds
-		ps := rmState.priorityLevelStates[fs.Spec.PriorityLevelConfiguration.Name]
+		if fs != nil {
+			flowDistinguisher := requestDigest.ComputeFlowDistinguisher(fs.Spec.DistinguisherMethod)
+			hashValue = hashFlowID(fs.Name, flowDistinguisher)
+			priorityLevelName = fs.Spec.PriorityLevelConfiguration.Name
+		}
```

Comment:
If we are going to allow `fs == nil` then there should be a flow identifier and a hash value computed in this case.

Reply:
refactored, please take another look

Reply:
The KEP specifies different default handling for requests that should be exempt vs requests that should not be exempt.